src/tfr_navigation/include/tfr_navigation/navigator.h

```diff
@@ -34,17 +36,24 @@ class Navigator
         void navigate(const tfr_msgs::NavigationGoalConstPtr &goal);
 
         void update_position(const nav_msgs::OdometryConstPtr &msg);
+        
+        void update_feedback();
+        void update_result();
 
-        using Server = actionlib::SimpleActionServer<tfr_msgs::NavigationAction>; 
 
-        ros::NodeHandle& node;
 
+        using Server = actionlib::SimpleActionServer<tfr_msgs::NavigationAction>; 
+        using NavStack =
+            actionlib::SimpleActionClient<move_base_msgs::MoveBaseAction>;
+        ros::NodeHandle& node;
+        //NOTE delegate initialization of mgr to ctor
+        NavigationGoalManager goal_manager;
         //NOTE delegate initialization of server to ctor
         Server server;                      //handle to as
+        //NOTE delegate initialization of server to ctor
+        NavStack nav_stack;
         //NOTE delegate initialization of subscriber to ctor
         ros::Subscriber odom_subscriber;    
-        //NOTE delegate initialization of mgr to ctor
-        NavigationGoalManager goal_manager;
 
         //needed msgs
         tfr_msgs::NavigationFeedback feedback{};
```

Comment:
As far as I can tell, you don't need to have `feedback` as a member variable.  You can just have `update_feedback` return a fresh object, since it's building it from scratch anyway.

This is a little better, I think, and it has the added benefit of removing any possibility that future edits to this class could run into race conditions if they were for some reason modifying this value.

Reply:
Same thing goes for the `result` message, I believe.

Reply:
Will do!

Reply:
Well actually looking at the code, after the new thread safety changes, I don't think we have any more race conditions. 

And, I don't think making that change provides good interface consistency with the current position method.

So either we leave it alone or we do a full scale refactor of the interface, and have update position return a different object.

Reply:
If that's the scope of the change you are suggesting that needs to be a seperate pull request.